app/views/exam_templates/_boot.js.erb

```diff
@@ -14,6 +14,74 @@ document.addEventListener("DOMContentLoaded", function () {
   });
 });
 
+document.addEventListener("DOMContentLoaded", () => {
+  createConsumer().subscriptions.create(
+    {channel: "ExamTemplatesChannel", course_id: <%= @current_course.id %>},
+    {
+      received(data) {
+        if (data["status"] !== null) {
+          const message_data = generateMessage(data);
+          renderFlashMessages(message_data);
+          if (data["status"] === "completed") {
+            handleJobCompletion(data);
+          }
+        }
+      },
+      connected() {},
+      disconnected() {},
+    }
+  );
+});
+
+function generateMessage(status_data) {
+  let message_data = {};
+  switch (status_data["status"]) {
+    case "queued":
+      message_data["notice"] = I18n.t("poll_job.queued");
+      break;
+    case "in_progress":
+      if (status_data["job_class"] === "SplitPdfJob") {
+        message_data["notice"] = I18n.t("poll_job.split_pdf_job", {
+          progress: status_data["page_number"],
+          total: status_data["total_pages"],
+          exam_name: status_data["exam_name"],
+        });
+      } else if (status_data["job_class"] === "GenerateJob") {
+        message_data["notice"] = I18n.t("poll_job.generate_job", {
+          progress: status_data["page_number"],
+          total: status_data["total_pages"],
+          exam_name: status_data["exam_name"],
+        });
+      }
+      break;
+    case "completed":
+      message_data["success"] = I18n.t("poll_job.completed");
+      break;
+    case "failed":
+      if (status_data["exception"]) {
+        message_data["error"] = I18n.t("job.status.failed.message", {
+          error: status_data["exception"],
+        });
+      } else {
+        message_data["error"] = I18n.t("job.status.failed.no_message");
+      }
+      break;
+  }
+  return message_data;
+}
+
+function handleJobCompletion(status_data) {
+  if (status_data["job_class"] === "SplitPdfJob") {
+    window.location.reload();
+  } else if (status_data["job_class"] === "GenerateJob") {
+    const path = download_generate_course_exam_template_path(
+      status_data["course_id"],
+      status_data["exam_id"]
+    );
+    window.location = `${path}?file_name=${status_data["file_name"]}`
```

Comment:
You should be able to pass in query parameters to the path helper method directly, rather than construct the query here using string manipulation. See https://github.com/railsware/js-routes?tab=readme-ov-file#usage